lib/spack/spack/test/cmd/load.py

```diff
@@ -28,15 +24,26 @@ def test_manpath_trailing_colon(
     manpath search path via a trailing colon"""
     install("mpileaks")
 
-    sh_out = load("--sh", "--only", "package", "mpileaks")
-    lines = sh_out.split("\n")
-    assert any(re.match(r"export MANPATH=.*:;", ln) for ln in lines)
+    if sys.platform == "win32":
+        bat_out = load("--bat", "--only", "package", "mpileaks")
+        lines = bat_out.split("\n")
+        assert any(re.match(r'set "MANPATH=.*:"', ln) for ln in lines)
+
+        os.environ["MANPATH"] = "/tmp/man;"
+
+        bat_out = load("--bat", "--only", "package", "mpileaks")
+        lines = bat_out.split("\n")
+        assert any(re.match(r'set "MANPATH=.*;/tmp/man;:', ln) for ln in lines)
```

Comment:
A few things:

* We don't need to be on `win32` to test the output of `spack load` (when the shell arg is provided). So the "if" could be removed here
* AFAIK `PATH` ending in `:` only has significance on Linux, so we only need to assert that there
* That being said, the fact that this is asserting that the line is `r'set "MANPATH=.*;/tmp/man;:'` is confusing to me: what does the `:` at the end of that string mean?

Reply:
> We don't need to be on win32 to test the output of spack load (when the shell arg is provided). So the "if" could be removed here

Good point.

Reply:
> That being said, the fact that this is asserting that the line is r'set "MANPATH=.*;/tmp/man;:' is confusing to me: what does the : at the end of that string mean?

That was a result of terminating the manpath with colon. I change that to use os.pathsep